version: '3.8'

services:
  traefik:
    image: traefik:v2.5
    command:
      - "--api.insecure=${TRAEFIK_API_INSECURE}"
      - "--providers.docker=${TRAEFIK_PROVIDERS_DOCKER}"
      - "--entrypoints.web.address=${TRAEFIK_ENTRYPOINTS_WEB_ADDRESS}"
      - "--entrypoints.websecure.address=${TRAEFIK_ENTRYPOINTS_WEBSECURE}"
      - "--certificatesresolvers.myresolver.acme.tlschallenge=true"
      - "--certificatesresolvers.myresolver.acme.email=${LETSENCRYPT_EMAIL}"
      - "--certificatesresolvers.myresolver.acme.storage=/letsencrypt/acme.json"
      - "--providers.file.directory=/etc/traefik/dynamic"
      - "--providers.file.watch=true"
    ports:
      - "${TRAEFIK_PORT}:80"
      - "443:443"
      - "${TRAEFIK_DASHBOARD_PORT}:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "./traefik/letsencrypt:/letsencrypt"
      - "./traefik/traefik.yml:/etc/traefik/traefik.yml"
      - "./traefik/dynamic.yml:/etc/traefik/dynamic/dynamic.yml"
    networks:
      - chatbot_pdfreader_network
    restart: unless-stopped

networks:
  chatbot_pdfreader_network:
    external: true
